### PR TITLE
Initial commit of MPI-only Laghos with Sedov blast wave experiment

### PR DESCRIPTION
**Pull request by wdhawkins**
This PR adds an mpi-only version of Laghos (https://github.com/CEED/Laghos) with a Sedov blast wave experiment.  

Although all of the experiments run successfully, FOM and success criteria are not working correctly. I'm currently looking into this.

### CODE REVIEW

File: repo/laghos/package.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    # See lib/spack/spack/build_systems/makefile.py
+    def check(self):
+        with working_dir(self.build_directory):
+            make("test", *self.build_targets)
+ 
+    def install(self, spec, prefix):
+        mkdirp(prefix.bin)
+        install("laghos", prefix.bin)
+
+    @run_after("install")
```
Comment (pearce8) [inline repo/laghos/package.py]: @scheibelp Is run_after("install") the correct way to copy data files after installing the binary?

Comment (scheibelp) [reply repo/laghos/package.py]: This should work fine. Is there a failure at this step?

Using `cache_extra_test_sources` assumes the files live in `.../stage/spack-src/data` and that you want them in `<install-prefix>/data`

Comment (wdhawkins) [reply repo/laghos/package.py]: I tested this on a fresh `benchpark` install.  It looks to me like the bit of code that I hoped would  copy the `data` directory to `install-prefix/data` actually copied it to `install-prefix/.spack/test/data`.

Comment (scheibelp) [reply repo/laghos/package.py]: Whoops sorry yes: you'll need to use `install_tree` then (which allows setting up a `src` and a `dst`):

```
install_tree("data", prefix.data)
```

should do it

Comment (wdhawkins) [reply repo/laghos/package.py]: Thanks @scheibelp! That fixed the problem.

**Review by pearce8 — APPROVED** [review]

**Pull request merged**